js/woo.js

```diff
@@ -0,0 +1,1465 @@
+                }
+            }
+            const networkCode = defaultNetworkPair['code'];
+            return [ networkizedCode, networkCode ];
+        } else {
+            // if it was not returned above according to defaults & options, then return the first network
+            const currencyItem = this.currencies[code];
+            const networks = currencyItem['networks'];
+            const networkKeys = Object.keys (networks);
+            for (let j = 0; j < networkKeys.length; j++) {
+                const networkCode = networkKeys[j];
+                const networkItem = networks[networkCode];
+                const networkizedCode = this.safeString (networkItem, 'chained_currency_code');
+                if (networkizedCode !== undefined) {
+                    return [ networkizedCode, networkCode ];
+                }
+            }
+        }
+        return undefined;
+    }
+
+    getCurrencyByNetworkizedCode (chainedCode) { // TODO: this method can be moved into base
+        const keys = Object.keys (this.currencies);
+        for (let i = 0; i < keys.length; i++) {
+            const key = keys[i];
+            const currencyItem = this.currencies[key];
+            const currencyCode = currencyItem['code'];
+            if (chainedCode === currencyCode) {
+                return chainedCode;
+            } else {
+                const networks = currencyItem['networks'];
+                const networkKeys = Object.keys (networks);
+                for (let j = 0; j < networkKeys.length; j++) {
+                    const networkKey = networkKeys[j];
+                    const networkItem = networks[networkKey];
+                    if (networkItem['chained_currency_code'] === chainedCode) {
+                        return currencyItem;
+                    }
+                }
+            }
+        }
+        return undefined;
+    }
+
+    objectMemberAt (objOrArray, position) { // TODO: can be in base
+        return objOrArray[Object.keys (objOrArray)[position]];
+    }
+
+    hasChildWithKeyValue (obj, targetKey, targetValue) {
+        const keys = Object.keys (obj);
+        for (let i = 0; i < keys.length; i++) {
+            const currentKey = keys[i];
+            const childMember = obj[currentKey];
+            const value = this.safeInteger (childMember, targetKey, undefined);
+            if (value === targetValue) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    sanitizestr (str) { // TODO: this method can be moved into base
+        str = str.replace (' ', '-');
+        str = str.replace ('.', '-');
+        str = str.replace ('_', '-');
+        return str.toLowerCase ();
+    }
+
+    detectChainSlug (type) { // TODO: this method can be moved into base
+        const type_S = this.sanitizestr (type);
+        const chainsSlugs = {};
+        chainsSlugs['BTC'] = [ 'btc', 'btc-chain', 'btc-network', 'bitcoin', 'bitcoin-chain', 'bitcoin-network' ];
```

Comment:
Where do all these slugs come from? If they're mentioned in the API docs (exchange-specific) – we can keep them, otherwise we have to remove them, cause they're not unified and not used by this exchange.

Reply:
Actually, that is a part of 'in-house' unified base method which I've mentioned. the specific exchange (lets say Woo in this case) have only one or two matching title from that list of slug, however, these slugs were picked and joined from different exchanges, and that's why I've marked it with "..maybe move this method in base..", where we can have a better method body, like:
```
detectChainSlug (slug) {
 chainsSlugs['BTC'] = [
    'btc', // binance, gateio, bitfinex
    'btc-chain', // kraken, xyz
    'bitcoin-chain', //phemex, 
 ....
```
etc, and so, in implementations of exchange classes, we can have just `this.detectChainSlug()` applied to any network related slug and it will be automatically covnerted into unified network-slug (i.e. BTC, ERC20 or whatever will be chosen by CCXT unified networkid for specific chain).

The second way is that we created individual parsing functions inside derived exchange classes, but it will be far from unified convenience i think.

Reply:
First, I totally agree about unification.

However, before unifying chain slugs, we need to define exactly what a chain slug is and why we need it. The users are supposed to supply unified networks to methods like fetchDepositAddress, withdraw, etc. We should also return unified networks in the related structures. How is chain slug applied to that? I think we should take the info from the preloaded currencies into the account.

Before we can actually unify it – we need to spend more time on the design of this unification. And since it involves a lot of exchanges – we should probably do that in a standalone PR. This exchange integration, however, should not depend on that, we can't really freeze it for a few weeks till we get to unifying networks everywhere. We have to make this code clean and mergeable without external dependencies first.

Reply:
I understand, you're right. I've tried now to make this individual exchange to be a self-isolated. About the slugs, let's talk about it in discord thread.